src/wasm/wasm-emscripten.cpp

```diff
@@ -143,7 +143,27 @@ inline void exportFunction(Module& wasm, Name name, bool must_export) {
   wasm.addExport(exp);
 }
 
+static bool needsDynCall(Signature sig) {
```

Comment:
how about `hasI64`? This function by itself doesn't decide if we need a dyncall or not (we may have passed a flag that says we never need any), it just identifies an i64 specifically IIUC.